File: spoj/2-PRIME1/solution_2.cpp
#include <iostream>

using namespace std;

void primeNumber(int, int);

int main()
{
    int t;
    cin >> t;
    for (int i = 0; i < t; i++)
    {
        int m, n;
        cin >> m;
        cin >> n;
        primeNumber(m, n);
        cout << "\n";
    }

    return 0;
}

void primeNumber(int start, int end)
{
    for (int i = start; i <= end; i++)
    {
        bool odd = i % 2 != 0;
        int n = odd ? 3 : 2;
        bool prime = i != 1;
        while (n * n <= i && prime)
        {
            prime = i % n != 0;
            n+=2;
        }

        if (prime)
            cout << i << "\n";
    }
}

File: spoj/2-PRIME1/solution_2.py
def prime_num(start, end):
    
    for x in range(start,end+1):
        odd = x % 2 != 0
        n = (2,3)[odd]
        prime = x != 1
        while(n*n <= x and prime):
            prime = x%n != 0
            n+=2
        if prime :
            print(x)


if __name__ == "__main__":
    t = int(input())
    for x in range(t):
        m,n = input().split(" ")
        m = int(m)
        n = int(n)
        prime_num(m,n)
        print()
